pkg/apis/cincinnati/v1alpha1/cincinnati_types.go

```diff
@@ -39,6 +39,12 @@ type CincinnatiSpec struct {
 	// specified, it takes precedence over GitHubOrg and GitHubRepo and disables
 	// retrieval of graph data from GitHub.
 	GraphDataImage string `json:"graphDataImage,omitempty"`
+
+	// CertConfigMapKey refers to the ConfigMap key that holds a CA cert to the registry
+	// Cincinnati will contact to build the upgrade graph.  The name of the
+	// ConfigMap holding the key comes from the field additionalTrustedCA
+	// in the image.config.openshift.io API.
+	CertConfigMapKey string `json:"certConfigMapKey"`
```

Comment:
Just confirming: is this not a predictable value? Is there not a way for us to either know it up-front or automatically figure it out?

Reply:
It's not predictable.  See the example from these docs: https://docs.openshift.com/container-platform/4.2/registry/configuring-registry-operator.html#registry-operator-config-resources-configmap_configuring-registry-operator.  What's good is that there will be only one configmap to read from, but there can be multiple keys.  They keys are normally a registry's hostname, so Cincinnati wouldn't know which one to pick.

Reply:
Wouldn't it just be the `Registry` field on the Cincinnati Spec?

Reply:
We can run into danger if we assume `Registry` equals the Image Registry key because the object is a configmap key, so any key is acceptable.  Therefore, I want to avoid equating the two because there is a potential bug here when a user sets Registry=my.local.registry and there happens to already be a key in this configmap with 'my.local.registry'.  If the configmap keys were gated in some way as to assure they were valid registries and working keys, then I would agree.

Reply:
The docs say "The key is the host name of a registry with the port for which this CA is to be trusted." That seems clear. I'm not following where the uncertainty is. It's a map of hostnames to CAs, and it seems like we should just look for the hostname we care about to appear in that map. Surely that's what the OCP image registry itself is doing, right?

Reply:
I'm not sure how OCP image registry uses this configmap.  My expectation is that the image registry uses key/value to allow you to reach external registry images through the internal registry address.

What were weighing here is the assumption that our valid registry string is equal to an unverified registry string, versus the user typing in a string.  Bottom line, we're using this API as if it were a configmap, so I'm not too thrilled about the idea of assuming key equivalence when the trade-off is small. 

Before landing on using this API, I was going to use a configmap we owned (we talked about this already).   Going with the API seemed like a better fit only because this looked like the general place folks would store this data.  That being said, I'd be fine with removing this field if we owned the data transmission format and could guarantee the relationship. 

Regardless, I think both sides of this is splitting hairs because we're talking about a small UX gain against major user error.  It's unusual, but I guess that makes me the user pessimist here.  Maybe the OTA team can decide the fate of this field, preferably outside of this PR. WDYT?

Reply:
I guess I'm thinking that >99% of the time the values are going to be the same. I guess we could later remove this field if we find that they're the same 100% of the time.

Do you have an example scenario in mind that might help me understand the concern better?

Reply:
Admin (a different ryanh) wants a place to store his external registry cert, so he creates an entry in the image config API with `key: ryanh`, that isn't a resolvable name from inside the pods.  Then, the admin gives me access to his cluster.  I happen to run my own registry located at `ryanh` and I make it so `ryanh` will resolve to my registry in the cluster.  If I go to run Cincinnati at this point, a number of things can go wrong depending what the Admin uses that cert for, whether I followed the docs or not, ect....

Reply:
I guess I'm not seeing the value in why someone would put a CA in there whose key isn't a valid hostname.

I'm good with merging this as-is, and let's check in next week with the image registry operator folks just to double-check what assumptions we can or can't make about the contents.

Reply:
> assumptions we can or can't make about the contents.
>
👍

Reply:
I needed to sleep on this.  The problem with my argument is, while yes it's true that we have to trust unverified keys, that's the same trust you extend to any configmap you consume. 

My perspective on this is driven by the fact that I don't trust we're using this API exactly the way it's intended or something is missing.  This description makes me think it's the right place: `AdditionalTrustedCA in the image.config.openshift.io resource to provide additional CAs that should be trusted when contacting external registries.`  But why would anyone that wants registry certs in a pod go through all this trouble creating a key and copying a configmap locally when they can just create a local configmap?   It just seems like the registry operator is incomplete.  We'll talk about this more next week, have a good weekend.